ssh/tailssh/tailssh.go

```diff
@@ -1448,6 +1454,53 @@ func (ss *sshSession) openFileForRecording(now time.Time) (_ io.WriteCloser, err
 	return f, nil
 }
 
+// monitorRecorderStatus watches the IPNBus for any netmap updates that indicate
+// that the recorder node is now inaccessible and signals via the returned
+// channel.
+func (ss *sshSession) monitorRecorderStatus(last *tailcfg.SSHRecordingAttempt) (<-chan struct{}, error) {
+	// identify the recorder node key by its tailscale IP
+	status := ss.conn.srv.lb.Status()
+	var recorderNodeKey key.NodePublic
+	if status != nil && len(status.Peers()) > 0 {
+		for nodeID, peer := range status.Peer {
+			if slices.Contains(peer.TailscaleIPs, last.Recorder.Addr()) {
+				recorderNodeKey = nodeID
+			}
+		}
+	}
+
+	// if we can't find the recorder error out.
+	if recorderNodeKey.IsZero() {
+		ss.logf("unable to find recorder node in status peer list")
+		return nil, errors.New("recorder node not found")
+	}
+
+	// watch indefinitely for IPN updates to observe if the recorder node ever goes offline.
+	recorderOffline := make(chan struct{})
+	go func() {
+		var mask ipn.NotifyWatchOpt
+		ss.conn.srv.lb.WatchNotificationsAs(ss.ctx, nil, mask, func() {}, func(roNotify *ipn.Notify) bool {
```

Comment:
optional, these are equivalent
```suggestion
		ss.conn.srv.lb.WatchNotifications(ss.ctx, mask, func() {}, func(roNotify *ipn.Notify) bool {
```